pkg/mounter/mounter-rclone.go

```diff
@@ -23,6 +28,73 @@ type rcloneMounter struct {
 	mountOptions  []string
 }
 
+const (
+	namespace  = "default"
+	secretName = "cos-s3-csi-pvc"
+)
+
+func updateMountOptions(dafaultMountOptions []string) ([]string, error) {
+	var config *rest.Config
+	if home := homedir.HomeDir(); home != "" {
+		klog.Infof("HOME: %s", home)
+	}
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		klog.Errorf("Error accessing config: %v\n", err)
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		klog.Errorf("Error accessing client: %v\n", err)
+		return nil, err
+	}
+
+	// Retrieve the Secret
+	secret, err := clientset.CoreV1().Secrets(namespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Error accessing Secret: %v\n", err)
+		return nil, err
+	}
+	mountOptsMap := make(map[string]string)
+
+	// Create map out of array
+	for _, e := range dafaultMountOptions {
+		opts := strings.Split(e, "=")
+		if len(opts) == 2 {
+			mountOptsMap[opts[0]] = opts[1]
+		}
+	}
+
+	stringData := string(secret.Data["mountOptions"])
+
```

Comment:
How do we handle scenario for  "mountOptions"   not provided

Reply:
addressed